### PR TITLE
Add Project Euler problem 79 solution 1

### PR DESCRIPTION
**Pull request by ishabahmed**
### Describe your change:

Added Project Euler problem 79 solution 1

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_079/sol1.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import itertools
+import os
+
+
+def solution() -> int:
+    """
+    Returns the shortest possible secret passcode of unknown length.
+
+    >>> solution()
+    73162890
```
Comment (dhruvmanila) [inline project_euler/problem_079/sol1.py:27]: As mentioned in the [README](https://github.com/TheAlgorithms/Python/tree/master/project_euler#coding-style):

> There should not be a doctest for testing the answer as that is done by our GitHub Actions build ...

```suggestion
```

Instead, have some test cases. Now as this requires reading a file to get the input, you could separate out the core part of the solution (which is everything after the file is read) and take in a parameter `logins`:

```python
# This function could be tested with variety of inputs.
def find_secret_passcode(logins):
	...


def solution():
	# Read the keylog.txt file and fill the `logins` array
    logins = [...]

	return find_secret_passcode(logins)
```

**Review by dhruvmanila — CHANGES_REQUESTED** [review]
It's better to keep the filename same as that of problem statement: `keylog.txt`. We already know the problem number from the directory.

File: project_euler/problem_079/sol1.py (hunk 1 above)
Comment (ishabahmed) [reply project_euler/problem_079/sol1.py:27]: Good idea. I will implement this.

File: project_euler/problem_079/sol1.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                < permutation.index(login[2])
+            ):
+                satisfied = False
+                break
+
+        if satisfied:
+            return int("".join(permutation))
+            break
+
+    return 0
```
[the remark this answers is not in the document]
Comment (ishabahmed) [reply project_euler/problem_079/sol1.py:49]: Given the fixed input from keylog.txt, the code should hypothetically never reach that line. However, it was necessary to include it to ensure that mypy detects a reachable return statement. Following the suggestion to split the core solution into a separate function, raising an exception makes sense, so I will implement that.

File: project_euler/problem_079/sol1.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    return find_secret_passcode(logins)
+
+
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
+
+    print(solution())
```
Comment (dhruvmanila) [inline project_euler/problem_079/sol1.py:76]: `doctest` are tested directly by the CI. If you want to run them locally, I would recommend to use the command:
```console
python -m doctest -v sol1.py
```

```suggestion
    print(f"{solution() = }")
```

**Review by dhruvmanila — APPROVED** [review]

**Review by dhruvmanila — APPROVED** [review]

**Pull request merged**